core/src/main/scala/com/rallyhealth/vapors/core/logic/LogicalConnective.scala

```diff
@@ -0,0 +1,143 @@
+package com.rallyhealth.vapors.core.logic
+
+import cats.{Invariant, Semigroupal}
+
+sealed trait LogicalConnective[A] {}
+
+object LogicalConnective {
+
+  @inline final def apply[A](implicit A: LogicalConnective[A]): LogicalConnective[A] = A
+}
+
+sealed trait UnaryConnective[A] extends LogicalConnective[A] {
+
+  def apply(value: A): A
+}
+
+trait Negation[A] extends UnaryConnective[A] {
+  @inline final def negation(value: A): A = apply(value)
+}
```

Comment:
Not to be confused with `Negative` used for inverting a positive number to a negative one... I should put that in the documentation.